src/streaming/streamingclient.py

```diff
@@ -0,0 +1,132 @@
+        if not access_token:
+            raise ValueError('access_token must be provided')
+
+        if not config:
+            raise ValueError('config must be provided')
+
+        self.access_token = access_token
+        self.config = config
+        self.base_url = base_url = 'wss://api.rev.ai/speechtotext/{}/stream'.format(version)
+        self.on_error = on_error
+        self.on_close = on_close
+        self.on_connected = on_connected
+        self.client = websocket.WebSocket(enable_multithread = True, 
+            on_error = self.on_error, 
+            on_close = self.on_close, 
+            on_connected = self.on_connected)
+
+    def start(self, generator):
+        """Function to connect thde websocket to the URL and start the response thread
+
+        :param generator: generator object that yields binary audio data
+        """
+        url = self.base_url + '?access_token={}'.format(self.access_token) \
+             + '&content_type={}'.format(self.config.get_content_type_string())
+        try:
+            self.client.connect(url)
+            self._start_send_data_thread(generator)
+            return self._responses()
+        except Exception as e:
+            self.client.abort()
+            self.on_error(e)
+
+    def end(self):
+        """Function to end the streaming service, close the websocket.
+        """
+        self.client.abort()
+
+    def _start_send_data_thread(self, generator):
+        """Function to send binary audio data from a generator with threading
+
+        :param generator: generator object that yields binary audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        if hasattr(self, 'request_thread'):
+            if self.request_thread.isAlive():
+                raise ValueError("""Data is still being sent and will interfere with the responses.""")
+
+        self.request_thread = threading.Thread(target = self._send_data, 
+                                args = [generator])
+        self.request_thread.start()
+
+    def _send_data(self, generator):
+        """Function used in a thread to send requests to the server.
+
+        :param generator: enerator object that yields binary audio data
+        """
+        if not generator:
+            raise ValueError('generator must be provided')
+
+        for chunk in generator:
+            self.client.send_binary(chunk)
+
+        self.client.send("EOS")
+
+    def _responses(self):
+        """A generator of reponses from the server. Yields the data decoded.
+        """
+        while True:
+            with self.client.readlock:
+                opcode, data = self.client.recv_data()
+            if six.PY3 and opcode == websocket.ABNF.OPCODE_TEXT:
+                dec_data = data.decode('utf-8') 
+                data_dict = json.loads(dec_data)
+                if data_dict['type'] == 'connected':
+                    yield self.on_connected(data_dict['id'])
```

Comment:
I think yielding this will either error since we're yeilding a void or will yield `None`. Instead we should call `self.on_connected` then pass. Python generators continue to run until a `yield` or `return` is encountered so if we just pass here the generator will keep looking at the responses until we get to a hypothesis message